src/Storages/HDFS/StorageHDFS.cpp

```diff
@@ -120,8 +120,15 @@ namespace
 
     std::pair<String, String> getPathFromUriAndUriWithoutPath(const String & uri)
     {
-        const size_t begin_of_path = uri.find('/', uri.find("//") + 2);
-        return {uri.substr(begin_of_path), uri.substr(0, begin_of_path)};
+        auto pos = uri.find("//");
+        if (pos != std::string::npos && pos + 2 != uri.end())
```

Comment:
`pos + 2` is position/index and `uri.end()` is iterator, it's incorrect to compare them